Odata-docs/webapi-8/fundamentals/query-options.md

```diff
@@ -0,0 +1,568 @@
+public IQueryable<Customer> Get()
+{
+    return Customers.AsQueryable<Customer>();
+}
+```
+
+## Config options vs AllowedQueryOptions
+When registering OData services, we add query options configurations as shown below:
+
+```csharp
+.AddOData(
+    options => options.Select().OrderBy().Expand().Count().SetMaxTop(null)
+)
+```
+
+These are global configurations. If a query option is not enabled here, it cannot be enabled in the controller.
+
+In the above example, we did not enable the $filter query option. If we configure a controller as shown below, the $filter will not be applied since we had not enabled $filter in the global configuration.
+
+```csharp
+[EnableQuery(AllowedQueryOptions = AllowedQueryOptions.Filter)]
+public IActionResult Get()
+{
+    return Ok(Customers);
+}
+```
+
+## Invoking query options directly
+There are scenarios where we are unable to use `EnableQuery` attribute e.g When inside the controller method, we are getting data from multiple sources and we would want that data to be processed before it's returned to the controller method.
+
+We use the `ODataQueryOptions.ApplyTo` method to apply the required query options.
+
+### SelectExpand query
+```csharp
+public IQueryable<Customer> Get(ODataQueryOptions<Customer> options)
+{
+    IQueryable results = options.ApplyTo(Customers.AsQueryable());
+
+    return results as IQueryable<Customer>;
+}
+```
+In the example above, `options.SelectExpand` property will be set when we make the request below:
+
+```http
+GET http://localhost:6285/odata/Customers?$expand=Orders
+```
+The `options.SelectExpand` property is an instance of `SelectExpandQueryOption`.
+
+Below will be the output:
+
+```json
+{
+    "@odata.context": "http://localhost:6285/odata/$metadata#Customers(Orders())",
+    "value": [
+        {
+            "Id": 1,
+            "Name": "Customer 1",
+            "Age": 31,
+            "Orders": [
+                {
+                    "Id": 1001,
+                    "Price": 10,
+                    "Quantity": 10
+                },
+                {
+                    "Id": 1002,
+                    "Price": 35,
+                    "Quantity": 2
+                }
+            ]
+        },
+        ...
+        ...
+    ]
+}
+```
+
+Alternatively we can call `ApplyTo` on the `SelectExpand` property as follows:
+
+```csharp
+public IQueryable<Customer> Get(ODataQueryOptions<Customer> options)
+{
+    IQueryable results = options.SelectExpand.ApplyTo(Customers.AsQueryable(), new ODataQuerySettings());
+
+    return results as IQueryable<Customer>;
+}
+```
+
+If we don't want to pass the query option in the request, but we want the `Orders` property to be expanded, we can initialize the `SelectExpandQueryOption` and set it in the request.
+
+```csharp
+public IQueryable<Customer> Get(ODataQueryOptions<Customer> options)
+{
+    options.Request.ODataFeature().SelectExpandClause = new SelectExpandQueryOption(null, "Orders", options.Context,
```

Comment:
I didn't know this. Good to know. Does this also override/overwrite the request query options if it was already part of the URL?

Reply:
Yes it will override